dokan/dokan.c

```diff
@@ -249,6 +249,12 @@ int DOKANAPI DokanMain(PDOKAN_OPTIONS DokanOptions,
     return DOKAN_MOUNT_ERROR;
   }
 
+  
+  // wait until the device is realy mounted
+  if (DokanOptions->Options & DOKAN_OPTION_MOUNT_MANAGER) {
+	  WaitForMount(instance);
```

Comment:
It is an interesting way to solve it :).
But as you said it looks like a hack. Wouldn't be better to handle this wait through [RegisterDeviceNotification](https://msdn.microsoft.com/fr-fr/library/windows/desktop/aa363431%28v=vs.85%29.aspx)  to be notified of FSRTL_VOLUME_MOUNT?